lib/supavisor/client_handler.ex

```diff
@@ -783,32 +797,49 @@ defmodule Supavisor.ClientHandler do
   end
 
   defp db_checkout(_, _, data) do
-    {time, db_pid} = :timer.tc(:poolboy, :checkout, [data.pool, true, data.timeout])
+    start = System.monotonic_time(:microsecond)
+    db_pid = :poolboy.checkout(data.pool, true, data.timeout)
     Process.link(db_pid)
+    db_sock = DbHandler.checkout(db_pid, data.sock, self())
     same_box = if node(db_pid) == node(), do: :local, else: :remote
-    Telem.pool_checkout_time(time, data.id, same_box)
-    {data.pool, db_pid}
+    Telem.pool_checkout_time(System.monotonic_time(:microsecond) - start, data.id, same_box)
```

Comment:
I recall that somewhere in the code I have created macro that runs something similar to this example. Maybe it would be useful to reuse such macro to make intent clearer there.

Reply:
NVM, I have written it, but it never landed.